web/vim_lib.js

```diff
@@ -72,6 +75,38 @@ var LibraryVIM = {
 
     dropbox: null,
 
+    // from https://github.com/s-macke/jor1k/blob/master/js/lib/utf8.js
+    // LICENSE: https://github.com/s-macke/jor1k/blob/master/LICENSE.md
+    UnicodeToUTF8Stream: function(key) {
```

Comment:
I remembered that there are utf8 related functions in emscripten js libararies. Have you checked out?